--- ../src-base/minecraft_server/net/minecraft/src/ConsoleCommandHandler.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src-work/minecraft_server/net/minecraft/src/ConsoleCommandHandler.java	0000-00-00 00:00:00.000000000 -0000
@@ -3,6 +3,8 @@
 import java.util.Iterator;
 import java.util.Set;
 import java.util.logging.Logger;
+
+import cpw.mods.fml.server.FMLServerHandler;
 import net.minecraft.server.MinecraftServer;
 
 public class ConsoleCommandHandler
@@ -41,7 +43,11 @@
                 int var9;
                 WorldServer var10;
 
-                if (var4.equalsIgnoreCase("save-all"))
+                if (FMLServerHandler.instance().handleServerCommand(var2, var6.getUsername(), var6))
+                {
+                    //We handled it
+                }
+                else if (var4.equalsIgnoreCase("save-all"))
                 {
                     this.func_22115_a(var7, "Forcing save..");
 
